algorithm.py

```diff
@@ -114,20 +123,24 @@ def transform_from_partially_directed(graph_G: PartiallyDirectedGraph):
         correction_for_vs = defaultdict(int)    # default value is zero
         for edge in g.es:
             edge["transformed"] = False
-            if edge["directed"] is False:
+            if not edge["directed"]:
                 correction_for_vs[edge.source] -= 1
                 correction_for_vs[edge.target] += 1
 
+        #print("DIR: {}".format(g.es['directed']))
+        #print("In: {}, Out: {}".format(g.degree(1, type="in"), g.degree(1, type="out")))
+        
         # greedy transformation of undirected edges
         for edge in g.es:
-            if edge["directed"] is False:
+            #print("dir: {}".format(edge['directed']))
+            if not edge["directed"]:
                 source_vertex_id = edge.source
                 target_vertex_id = edge.target
 
                 # degree = incoming - outgoing + correction
                 source_vertex_degree = g.degree(source_vertex_id, type="in") - g.degree(source_vertex_id, type="out") \
                                        + correction_for_vs[source_vertex_id]
-                if source_vertex_degree > 0:
```

Comment:
po tej zmianie test test_transforming_g_to_g1 nie przechodzi

Reply:
Wiec było tak że liczyło `source_vertex_degree`, wychodziło +1 czyli wchodził do if `source_vertex_degree > 0`.
Usuwało krawędź wychodzącą od `source_vertex` dla którego wyszło +1 i dodawało krawędź która wchodzi do tego wierzchołka. Czyli czyli jeśli dobrze pamiętam to wchodząca jest z plusem więc zwiększało `source_vertex_degree`
Chyba powinna być tak skierowana by to zmniejszać

Reply:
No spoko tylko przy tej konwencji trzeba zamienić na odwrotną logikę tworzenia `correction_for_vs`

Reply:
Bo w przypadku tego wychodzi że stopień jest ujemny czyli +1.
![image](https://user-images.githubusercontent.com/42911623/82842166-cf85d300-9ed8-11ea-8a05-4744808624d3.png)

Reply:
Dobra masz rację pomyliłem się. Zapomniałem o correction. Teraz to nawet bez correction powinno być ok, bo krawędź nieskierowaną którą badamy zawsze biblioteka policzy jako wychodzącą?

Reply:
Nie wiadomo jak ją policzy, to jest niedeterministyczne.

Reply:
Nie jest tak że tworząc graf biblioteka skieruje od niższego indeksu wierzchołka do wyższego? Więc zawsze badając ten wierzchołek, który ma niższy indeks, krawędź nieskierowana będzie wychodząca?